### PR TITLE
Add minimal setup for linting Python code

### PR DESCRIPTION
**Pull request by fredericomba**
This pull requests add the minimum setup for automatically checking if the code is violating some style. Unfortunately, it seems that the Python linters out there do not have a rule for checking if this code style is being used:

```Python
def example(chosen_input: int):
  if chosen_input: # wrong, implicit check for non-zero integer
     pass
```

I've checked the [linting rules provided by Ruff](https://docs.astral.sh/ruff/rules/), the fastest Python linter out there, and it seems to not have support for this.

### CODE REVIEW

**Review by MuhammadMuradG — CHANGES_REQUESTED** [review]
@fredericomba I don't understand the purpose of this PR, could you please explain more? I think what you are submitted is more relating to setup a linting to your python environment.

File: requirements.txt [hunk 1]
```diff
@@ -7,3 +7,4 @@ pycparser==2.21
 PyQt5==5.15.10
 PyQt5-Qt5==5.15.2
 PyQt5-sip==12.13.0
+ruff==0.3.5
```
Comment (MuhammadMuradG) [inline requirements.txt:10]: I think we don't need to require from the user to install this before run the code.

Comment (MuhammadMuradG) [reply requirements.txt:10]: You are free to choose whether to continue work on this issue or close it, it is your decision. If you think it's best to close it out, we can close it. Feel free to open a new issue for any features you're excited about and thanks for your work on this so far!

Comment (Yuri-SVB) [reply requirements.txt:10]: Devs, present the opposing alternatives to me in a succinct, objective and technical way, so I decide on them.

Comment (MuhammadMuradG) [reply requirements.txt:10]: @Yuri-SVB you can find a brief description of opposing alternatives listed in the related issue #75.

Comment (Yuri-SVB) [reply requirements.txt:10]: We'll continue that on issue #75.

**Pull request closed without merging**